.github/workflows/icc-test-image.yaml

```diff
@@ -0,0 +1,26 @@
+name: Build in-cluster configuration test Image
```

Comment:
we will get many more tests that need an image. Worth to make this generic and have multple binaries in it?

Reply:
Maybe we start by renaming this from icc-test-image to kcp-test-image (or something more generic like that), but save multiple binaries/multiple subcommands for a follow-up when we have a need for them?